### ISSUE TITLE
Check `MsgChannelUpgradeInit` is signed by authority

### ISSUE THREAD

**Issue by crodriguezvega** [cosmos/ibc-go#4186]
_Reference: https://github.com/cosmos/ibc/pull/999_

Ideally we work on this after https://github.com/cosmos/ibc-go/pull/4160 is merged, since that PR will bring [the addition of `authority` to the core IBC keeper](https://github.com/cosmos/ibc-go/blob/0b746b68c58450d2e60c7a01ac349da8563b42b0/modules/core/keeper/keeper.go#L37), and then the change would be quite simple:

- In [`ChanUpgradeInit` rpc handler](https://github.com/cosmos/ibc-go/blob/e00fb0fda1c3fce67a2e18656bef6f4f67577aed/modules/core/keeper/msg_server.go#L699), add at the top:

```golang
if k.GetAuthority() != msg.Signer {
  return nil, errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Signer)
}
```

**Comment by charleenfei** [comment]
I started working on this issue, but ran into complications while testing.

Basically, the way we currently run tests for the `msg_server` code relies on us passing the msg into the `suite.chainA.SendMsgs(msg)` function, which will call `simapp.SignAndDeliver`. However, the simapp function requires the private key of the signer in order to generate the mock signed tx. We typically grab from the `TestChain` struct because we have access to it as the sender account is [generated from a private key which is custom generated for this use case and then saved in the `TestChain` struct. ](https://github.com/cosmos/ibc-go/blob/main/testing/chain.go#L110)

However, in this testing case we set up the IBC keeper authority in the `app.go` to the gov module, but we do not have access to the private key for this account. Therefore, the test will fail when at the `SignAndDeliver` step because the private key will not match the sender account address. 

one way around this would be to update the `MsgChannelUpgradeInit` to follow how `MsgUpdateParams` is set up, so that `authority` is a separate field on the msg from `msg.Signer`, and then we rely on the SDK [proto message option extension](https://github.com/cosmos/cosmos-sdk/pull/10977) to signal that the msg signer should be the IBC keeper authority, and have the check in the msg server on `MsgChannelUpgradeInit` [just check against `msg.Authority` ](https://github.com/cosmos/ibc-go/blob/main/modules/core/keeper/msg_server.go#L702) rather than checking against the msg.Signer field. The test would then look like [the test for `UpdateClientParams`](https://github.com/cosmos/ibc-go/blob/48a7baf57653ebb94c575f9e528dcc9f4a979fdd/modules/core/keeper/msg_server_test.go#L787).

**Comment by damiannolan** [comment]
We can modify the endpoint helper: `Endpoint.ChanUpgradeInit` to instead of routing the `MsgChannelUpgradeInit` to the application, route a new gov proposal containing the `MsgChannelUpgradeInit`.

Within the endpoint helper we can vote on the proposal and ensure it passes and executes the msg before returning.

**Comment by crodriguezvega** [comment]
Closed by #4773
